grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/gradle/dependencies/MavenInstallArtifactsCalculator.kt

```diff
@@ -0,0 +1,212 @@
+        return filterDependencies(variantDependencies)
+    }
+
+    /**
+     * Takes a list of variants and the list of configurations in them to produce a `MavenExternalArtifact`
+     * The data required for `MavenExternalArtifact` comes from different places and this method merges
+     * from all of them to produce `Map` of variants and `MavenExternalArtifact`s.
+     *
+     *  * Repository is calculated from merging all repositories in the project.
+     *  @see [RepositoryDataSource.allRepositoriesByName]
+     *  * Exclude rules are calculated from `ExternalDependency` provided from `configuration.dependencies`
+     *  * [ResolutionResult] is used to calculate dependency versions to ensure final version after
+     *  dependency resolution is used
+     *
+     */
+    private fun resolveVariantDependencies(
+        variantConfigs: Map<String, List<Configuration>>
+    ): Map<String, List<MavenExternalArtifact>> {
+        val repositories = repositoryDataSource.allRepositoriesByName
+        return variantConfigs.mapValues { (_, configurations) ->
+            val excludeRules = calculateExcludeRules(configurations)
+            configurations
+                .asSequence()
+                .filter { it.isCanBeResolved }
+                .map { it.incoming }
+                .flatMap { resolvableDependencies ->
+                    try {
+                        resolvableDependencies
+                            .resolutionResult
+                            .root
+                            .dependencies
+                            .asSequence()
+                            .filterIsInstance<DefaultResolvedDependencyResult>()
+                            .map { it.selected }
+                            .filter { !it.toString().startsWith("project :") }
+                            .filterIsInstance<DefaultResolvedComponentResult>()
+                            .map { componentResult ->
+                                val version = componentResult.moduleVersion!!
+                                MavenExternalArtifact(
+                                    group = version.group,
+                                    version = version.version,
+                                    name = version.name,
+                                    repository = Repository(
+                                        name = componentResult.repositoryName!!,
+                                        repository = repositories[componentResult.repositoryName!!]!!
+                                    ),
+                                    componentResult = componentResult,
+                                    excludeRules = excludeRules.getOrDefault(
+                                        version.toString(),
+                                        emptyList()
+                                    )
+                                )
+                            }
+                    } catch (e: Exception) {
```

Comment:
Should we print out the stack trace? or any logs?
It might help for debugging later if there is an exception occurred

Reply:
Yeah agreed.